### ISSUE TITLE
Precompile

### ISSUE THREAD

**Issue by jondea** [jlapeyre/LambertW.jl#9]
What is the status of precompilation on LambertW.jl?

At the top of [src/LambertW.jl](https://github.com/jlapeyre/LambertW.jl/blob/b9fa5810905d713bb6a9ab4febcb1a77b902bd7f/src/LambertW.jl#L1-L2) it says this:
```julia
# with precompilation, tests cause segfault somehow due to BigFloat
#VERSION >= v"0.4.0-dev+6521" && __precompile__()
```
If I comment it out, my code works fine (I'm using Float64s on Julia 0.6). But as the comment suggests, the tests fail for BigFloat:
```julia-repl
julia> Pkg.test("LambertW")
INFO: Testing LambertW
WARNING: @test_approx_eq is deprecated, use `@test lambertw(z, k) ≈ res` instead
Stacktrace:
 [1] depwarn(::String, ::Symbol) at ./deprecated.jl:70
 [2] @test_approx_eq(::ANY, ::ANY) at ./deprecated.jl:1061
 [3] include_from_node1(::String) at ./loading.jl:576
 [4] include(::String) at ./sysimg.jl:14
 [5] include_from_node1(::String) at ./loading.jl:576
 [6] include(::String) at ./sysimg.jl:14
 [7] process_options(::Base.JLOptions) at ./client.jl:305
 [8] _start() at ./client.jl:371
while loading /home/jondea/.julia/v0.6/LambertW/test/lambertw_test.jl, in expression starting on line 58

signal (11): Segmentation fault
while loading /home/jondea/.julia/v0.6/LambertW/test/lambertw_test.jl, in expression starting on line 103
__gmpn_rshift at /usr/local/julia/bin/../lib/julia/libgmp.so (unknown line)
mpfr_add_ui at /buildworker/worker/package_linux64/build/deps/srccache/mpfr-3.1.5/src/add_ui.c:51
unknown function (ip: 0x7ffe605f509f)
Allocations: 2274762 (Pool: 2273538; Big: 1224); GC: 2
==============================[ ERROR: LambertW ]===============================

failed process: Process(`/usr/local/julia/bin/julia -Cx86-64 -J/usr/local/julia/lib/julia/sys.so --compile=yes --depwarn=yes --check-bounds=yes --code-coverage=none --color=yes --compilecache=yes /home/jondea/.julia/v0.6/LambertW/test/runtests.jl`, ProcessSignaled(11)) [0]

================================================================================
ERROR: LambertW had test errors
```

For a low level mathematical package, it would be very useful to have precompilation out of the box.

**Comment by jlapeyre** [comment]
This is a case of a more general issue that is still unresolved. For example https://github.com/JuliaLang/julia/issues/15722

**Comment by jlapeyre** [comment]
This is fixed with e99397e7d41ee97f040122be2fcd9a54e9254b08
The package is now precompiled.
A few other things are fixed as well.
